src/units/unit.cpp

```diff
@@ -2120,9 +2236,13 @@ void unit::apply_builtin_effect(std::string apply_to, const config& effect)
 	} else if(apply_to == "remove_ability") {
 		if(const config& ab_effect = effect.child("abilities")) {
 			for(const config::any_child ab : ab_effect.all_children_range()) {
-				remove_ability_by_id(ab.cfg["id"]);
+				const config& filter = ab_effect.child(ab.key);
```

Comment:
in this part tag of abilities used for filttered individualyy each abilities to remove(with id) or other attrbute:
example `[chance_to_hit]
overwrite_specials=one_side
apply_to=opponent
[/chance_to_hit]
[chance_to_hit]
overwrite_specials=both_sides
apply_to=opponent
[/chance_to_hit]
[chance_to_hit]
overwrite_specials=one_side
apply_to=defender
[/chance_to_hit]
[chance_to_hit]
overwrite_specials=both_sides
apply_to=defender
[/chance_to_hit]`

the wtriting is heavy, this why i added afilter_abilities below. 

and it is not ab.child because not compilabe, this must be here ab_effect.chil(ab.key)

Reply:
What if you removed the `const` on the line above? Would `ab.child` work then?

Reply:
no, i  have 'src\units\unit.cpp|2240|error: 'struct config::any_child' has no member named 'child'|

Reply:
My mistake. Looks like it's supposed to be `ab.cfg`, not `ab.child`. I should double-check the correct name before I make suggestions…

Reply:
changed